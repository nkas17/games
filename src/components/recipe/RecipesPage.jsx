import React from 'react';
import PropTypes from 'prop-types';
import toastr from 'toastr';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import RecipeList from './RecipeList';
import * as recipeActions from '../../actions/recipeActions';

/**
 * Page that has all the recipe things
 */
class RecipePage extends React.Component { //eslint-disable-line
	constructor(props, context) {
		super(props, context);

		this._redirectToAddRecipePage = this._redirectToAddRecipePage.bind(this);
		this._deleteRecipe = this._deleteRecipe.bind(this);
	}
	_redirectToAddRecipePage() {
		this.props.history.push('/recipe/new');
	}

	_deleteRecipe(recipeId) {
		console.log(`deleted recipe id - ${recipeId}`);

		event.preventDefault();

		this.props.actions.deleteRecipe(recipeId)
			.then(() => toastr.success('Recipe deleted'))
			.catch(error => toastr.error(error));
	}

	render() {
		const { recipes } = this.props;
		return (
			<div className="jumbotron">
				<h2>recipes</h2>
				<hr />
				<RecipeList
					recipes={recipes}
					deleteRecipe={this._deleteRecipe}
				/>
				<input
					type="submit"
					value="add recipe"
					className="btn btn-primary"
					onClick={this._redirectToAddRecipePage}
				/>
			</div>
		);
	}
}

RecipePage.propTypes = {
	recipes: PropTypes.arrayOf(PropTypes.object).isRequired,
	history: PropTypes.objectOf(PropTypes.any).isRequired,
	actions: PropTypes.objectOf(PropTypes.func).isRequired,
};

const mapStateToProps = state => ({
	recipes: state.recipes,
});

const mapDispatchToProps = dispatch => ({
	actions: bindActionCreators(recipeActions, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(RecipePage);
