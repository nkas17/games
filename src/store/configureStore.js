import {createStore, applyMiddleware, compose} from 'redux';
import reduxImmutableStateInvariant from 'redux-immutable-state-invariant' // eslint-disable-line
import thunk from 'redux-thunk';
import rootReducer from '../reducers';

const configureStore = (initialState) => {
	// Add wrapper to enable redux dev tools
	// @see https://github.com/zalmoxisus/redux-devtools-extension#12-advanced-store-setup

	const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

	return createStore(
		rootReducer,
		initialState,
		composeEnhancers(applyMiddleware(thunk, reduxImmutableStateInvariant()))
	);
}

export default configureStore;