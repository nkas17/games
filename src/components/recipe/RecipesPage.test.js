import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import RecipesPage from './RecipesPage';
import configureStore from '../../store/configureStore';

/* eslint-disable no-undef */

const setup = () => (
	{
		recipes: [
			{
				id: 'tacos',
				title: 'tacos',
				category: 'other',
				description: 'Tasty homemade tacos - hard or soft',
				ingredients: [
					{
						id: 'cheese',
						quantity: '8',
						unit: 'oz',
					},
					{
						id: 'ground beef',
						quantity: '1.5',
						unit: 'lbs',
					},
					{
						id: 'tortilla',
						quantity: '1',
						unit: '',
					},
				],
			},
			{
				id: 'quiche',
				title: 'quiche',
				category: 'other',
				description: 'Delicious Spinach Quiche and Ham & Cauliflower quiche',
				ingredients: [
					{
						id: 'cheese',
						quantity: '8',
						unit: 'oz',
					},
					{
						id: 'pie-crust',
						quantity: '2',
						unit: '',
					},
					{
						id: 'egg',
						quantity: '7',
						unit: '',
					},
				],
			},
		],
	}
);

const store = configureStore(setup());
const renderWithRouter = node => <Router>{node}</Router>;
const renderWithStore = node => <Provider store={store}>{node}</Provider>;

describe('RecipesPage', () => {
	it('renders correctly', () => {
		const tree = renderer.create(
			renderWithStore(
				renderWithRouter(
					<RecipesPage
						history={{}}
					/>))).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
