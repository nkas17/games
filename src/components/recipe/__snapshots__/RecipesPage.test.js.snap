// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipesPage renders correctly 1`] = `
<div
  className="jumbotron"
>
  <h2>
    recipes
  </h2>
  <hr />
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
           
        </th>
        <th>
          title
        </th>
        <th>
          description
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
           
        </td>
        <td>
          <a
            href="recipe/tacos"
            onClick={[Function]}
          >
            tacos
          </a>
        </td>
        <td>
          Tasty homemade tacos - hard or soft
        </td>
      </tr>
      <tr>
        <td>
           
        </td>
        <td>
          <a
            href="recipe/quiche"
            onClick={[Function]}
          >
            quiche
          </a>
        </td>
        <td>
          Delicious Spinach Quiche and Ham & Cauliflower quiche
        </td>
      </tr>
    </tbody>
  </table>
  <input
    className="btn btn-primary"
    onClick={[Function]}
    type="submit"
    value="add recipe"
  />
</div>
`;
